src/csharp/Microsoft.Spark.Worker/Processor/PayloadProcessor.cs

```diff
@@ -79,14 +65,22 @@ internal Payload Process(Stream stream)
             TaskContextHolder.Set(payload.TaskContext);
 
             payload.SparkFilesDir = SerDe.ReadString(stream);
+            SparkFiles.SetRootDirectory(payload.SparkFilesDir);
 
-            if (Utils.SettingUtils.IsDatabricks)
+            if (SettingUtils.IsDatabricks)
             {
                 SerDe.ReadString(stream);
                 SerDe.ReadString(stream);
             }
 
             payload.IncludeItems = ReadIncludeItems(stream);
+
+            // Register additional assembly handlers after SparkFilesDir has been set
```

Comment:
Maybe it's better to move few lines up to the place where SparkFilesDir is read.

Reply:
Moved.